### DISCUSSION TITLE
Different Namespaces in an OL event

### DISCUSSION THREAD

**Discussion by MMoussabbih** [OpenLineage/OpenLineage#1775] [Q&A]
Hello everyone,

I'm using the **OpenLineage JAR** for **Spark integration** and I noticed that **namespace** used in `"inputs"` is not the same as what I configured in `spark.openlineage.namespace: "new-test"`, input dataset is created in `"file"` namespace. Here is the event : 

>
    "eventType": "COMPLETE",
    "eventTime": "2023-04-14T07:37:51.376Z",
    "run": {
      "runId": "f560e90b-131f-4d14-8b35-f3262b323995",
      "facets": {
        "spark.logicalPlan": {
          "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
          "_schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunFacet",
          "plan": [
            {
              "class": "org.apache.spark.sql.catalyst.plans.logical.Aggregate",
              "num-children": 1,
              "groupingExpressions": [],
              "aggregateExpressions": [
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.Alias",
                    "num-children": 1,
                    "child": 0,
                    "name": "count(1)",
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 20,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": [],
                    "nonInheritableMetadataKeys": []
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression",
                    "num-children": 1,
                    "aggregateFunction": 0,
                    "mode": {
                      "object": "org.apache.spark.sql.catalyst.expressions.aggregate.Complete$"
                    },
                    "isDistinct": false,
                    "resultId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 15,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    }
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.aggregate.Count",
                    "num-children": 1,
                    "children": [
                      0
                    ]
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.Literal",
                    "num-children": 0,
                    "value": "1",
                    "dataType": "integer"
                  }
                ]
              ],
              "child": 0
            },
            {
              "class": "org.apache.spark.sql.catalyst.plans.logical.Project",
              "num-children": 1,
              "projectList": [],
              "child": 0
            },
            {
              "class": "org.apache.spark.sql.execution.datasources.LogicalRelation",
              "num-children": 0,
              "relation": null,
              "output": [
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "NAME",
                    "dataType": "string",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 0,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ],
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "ORIGIN_NAME",
                    "dataType": "string",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 1,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ],
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "value",
                    "dataType": "long",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 2,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ]
              ],
              "isStreaming": false
            }
          ]
        },
        "spark_version": {
          "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
          "_schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunFacet",
          "spark-version": "3.3.0",
          "openlineage-spark-version": "0.22.0"
        }
      }
    },
    "job": {
      "namespace": "new-test",
      "name": "test.adaptive_spark_plan",
      "facets": {}
    },
    "inputs": [
      {
        "namespace": "file",
        "name": "/Users/mac-35535/Documents/data/inputs/testdata",
        "facets": {
          "dataSource": {
            "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
            "_schemaURL": "https://openlineage.io/spec/facets/1-0-0/DatasourceDatasetFacet.json#/$defs/DatasourceDatasetFacet",
            "name": "file",
            "uri": "file"
          },
          "schema": {
            "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
            "_schemaURL": "https://openlineage.io/spec/facets/1-0-0/SchemaDatasetFacet.json#/$defs/SchemaDatasetFacet",
            "fields": [
              {
                "name": "NAME",
                "type": "string"
              },
              {
                "name": "ORIGIN_NAME",
                "type": "string"
              },
              {
                "name": "value",
                "type": "long"
              }
            ]
          }
        },
        "inputFacets": {}
      }
    ],
    "outputs": [],
    "producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
    "schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunEvent"

Do you have any idea about how we can have the **same namespace** for both `"job"` and `"inputs"` ?

Thanks in advance.

**Comment by pawel-big-lebowski** [comment]
Hi @MMoussabbih , 

Please take a look at: 
 * https://openlineage.io/blog/whats-in-a-namespace#dataset-namespaces
 * https://openlineage.io/blog/whats-in-a-namespace#job-namespaces

which clarify difference between dataset and job namespaces.  
Let us know if this answers your question.
